core/src/main/java/org/testcontainers/utility/ResourceReaper.java

```diff
@@ -125,7 +135,21 @@ public void onNext(Frame frame) {
 
         ContainerState containerState = new ContainerState() {
 
-            final InspectContainerResponse inspectedContainer = client.inspectContainerCmd(ryukContainerId).exec();
+            // inspect container response might initially not contain the mapped port
+            final InspectContainerResponse inspectedContainer = await()
+                .atMost(5, TimeUnit.SECONDS)
+                .pollInterval(new DynamicPollInterval(Duration.ofMillis(10)))
+                .pollInSameThread()
+                .until(
+                    () -> client.inspectContainerCmd(ryukContainerId).exec(),
+                    inspectContainerResponse -> inspectContainerResponse
```

Comment:
code style: it would be better to use a non-short lambda style here (`arg -> {\n return ...;\n }`)

Reply:
Just on the second lambda? And why?

Reply:
if we were to define the rule (and that's how it is defined in e.g. Prettier Java):
1. short lambda if one line
2. full lambda if multi-line

Just makes it easier to read/follow IMO.

Reply:
Alright, sensible rule and will disable the corresponding IntelliJ inspection to stop triggering me 😅

Reply:
Right! I almost forgot that there is such inspection, as I always disable it 😂